src/ir/selector.h

```diff
@@ -17,19 +16,11 @@ namespace raksha::ir {
 // For now, however, we only have implemented the FieldSelector portion.
 class Selector {
  public:
-
-  // A factory creating a Selector from a Manifest proto selector. Currently
-  // only implements the FieldSelector proto.
-  static Selector CreateFromProto(arcs::AccessPathProto_Selector selector) {
-    if (selector.has_field()) {
-      return Selector(FieldSelector(selector.field()));
-    }
-    LOG(FATAL) << "Found a Selector with an unimplemented specific type.";
-  }
-
   explicit Selector(FieldSelector field_selector)
     : specific_selector_(std::move(field_selector)) {}
 
+  std::variant<FieldSelector> variant() const { return specific_selector_; }
```

Comment:
I'm not wild about exposing the `std::variant` to the outside world, as it breaks the feel of this class being like an interface-style abstract class. Could we make the encode method be a `friend` of this class somehow so we didn't have to expose this everywhere?

Reply:
Why are you against exposing the underlying types? Don't you think we will need to create visitors over such variants outside of the class? Encode and Decode are just one instance, but I think we will add more? 

Also, isn't it similar to exposing the types of classes through a visitor pattern?

Reply:
See other conversation, this seems fine.